File: who-likes-it/who-likes-it.py
def likes(names):
    res = ""
    match names:
        case []:
            res = "no one likes this"
        case [x]:
            res = f"{x} likes this"
        case [x, y]:
            res = f"{x} and {y} like this"
        case [x, y, z]:
            res = f"{x}, {y} and {z} like this"
        case [x, y, *rest]:
            res = f"{x}, {y} and {len(rest)} others like this"
    return res

File: who_likes_it/who_likes_it.pl
likes([], "no one likes this").

likes([X], Result) :-
    sformat(Result, "~w likes this", [X]).

likes([X, Y], Result) :-
    sformat(Result, "~w and ~w like this", [X, Y]).
    
likes([X, Y, Z], Result) :-
    sformat(Result, "~w, ~w and ~w like this", [X, Y, Z]).

likes([X, Y | Tail], Result) :-
    length(Tail, TailSize),
    sformat(Result, "~w, ~w and ~d others like this", [X, Y, TailSize]).